### ISSUE TITLE
Have exercises that expect non-class implementations?

### ISSUE THREAD

**Issue by dsudia** [exercism/xtypescript#38]
@masters3d @kytrinyx 

I'm looking (and working) through the existing exercises and most of the ones I've looked at so far expect a class or object. One of the stereotypes about TS I've experienced in the wild is that you can't write JS the way you want to anymore, you have to basically write C# in JS. Or "TS is just so OO programmers can feel ok writing JS." Of course that's not true, for example VS Code being written in 100% functional code in TS.

I feel like having some tests expect a function might be helpful in showing for devs new to the language know you don't have to do everything in a class.

I can also see the argument for keeping all the test expectations consistent.

Thoughts?

**Comment by kytrinyx** [comment]
@dsudia From my perspective, Exercism is about teaching people about the basics of a language. I don't want to have consistency just for the sake of consistency. I think it would be very valuable to have exercises that take the various approaches that are idiomatic for the language.

TL;DR: Yes :) I think you're right that we should have some tests that expect a function.

**Comment by dsudia** [comment]
Ok cool. I think the focus for now can be getting more exercises in and sorted by difficulty. Once the presentation order is fairly set some refactoring can be done to get a good mix of syntaxes.

**Comment by kytrinyx** [comment]
@dsudia That sounds good

**Comment by masters3d** [comment]
@dsudia We dont always have to use classes; we can also use plain old JS objects but I do think we should wrapped them as a type `type Alias = { num: number } 
`
I like the idea of having different types of syntax as the difficulty increases.
